File: src/plugin/command/owner/UpdateCommandPlugin.java
package plugin.command.owner;

import com.rs.cores.CoresManager;
import com.rs.game.entity.actor.player.Player;
import com.rs.game.plugin.type.CommandPlugin;
import plugin.command.CommandManifest;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 9/1/2017
 */
@CommandManifest(description = "Updates the server after x seconds", types = { Integer.class})
public class UpdateCommandPlugin extends CommandPlugin {
	
	@Override
	public void handle(Player player, String[] args, boolean console, boolean clientCommand) {
		int delay = intParam(args, 1);
		CoresManager.safeShutdown(delay);
	}
	
	@Override
	public String[] identifiers() {
		return arguments("update", "shutdown");
	}
}

File: src/main/java/plugin/command/owner/UpdateCommandPlugin.kt
package plugin.command.owner

import engine.SystemManager
import game.content.plugin.type.CommandPlugin
import game.entity.actor.player.Player
import plugin.command.CommandManifest

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 9/1/2017
 */
@CommandManifest(description = "Updates the server after x seconds", types = [Int::class])
class UpdateCommandPlugin : CommandPlugin() {
    override fun handle(player: Player, args: Array<String>, console: Boolean, clientCommand: Boolean) {
        val delay = intParam(args, 1)
        SystemManager.safeShutdown(delay)
    }

    override fun identifiers(): Array<String> {
        return arguments("update", "shutdown")
    }
}